programs/survival/bridge.sc

```diff
@@ -0,0 +1,49 @@
+global_bridge_item = 'golden_sword';
+
+__can_bridge(entity) -> 
+(
+	if (entity ~ 'gamemode' == 'adventure', return(false));
+	if (!(entity ~ 'sneaking'), return(false));
+	holds_mainhand = query(entity, 'holds', 'mainhand');
+	holds_offhand = query(entity, 'holds', 'offhand');
+	if (!holds_mainhand || !holds_offhand, return(false));
+	if (!(holds_mainhand:0 ~ global_bridge_item), return(false));
+	true
+);
+
+__get_direction_vector(face) ->
```

Comment:
there is already a function that does that natively that you can use, `pos_offset`:
https://gnembon.github.io/scarpet/carpet/script/CarpetExpression.html#pos_offset
(sorry link won't work, you need to search)